.gitignore

```diff
@@ -61,3 +61,6 @@ vignettes/*.pdf
 
 # R Environment Variables
 .Renviron
+
+#slurm files on hpc
+/slurm/
```

Comment:
I have been git tracking these files. It may not make sense to track the tar_make scripts because they're short, but the #122 scripts we'll want to have tracked. Can you remove these from the .gitignore?

Reply:
I just looked at the files on Caldera and updated to get the correct plot

Reply:
oh, oops!

Reply:
No problem! The names of those files are not the most intuitive

Reply:
The result is about the same. Do you know how many nested gages are in each region? That could be a helpful number to have for each region

Reply:
The regions are in the `cluster_table`. There are different regions for high and med-high, so a total of 10 regions. 

> With a nestedness threshold of 0.5, there are a total of only 91 nested gages in the whole CONUS and there are no large groups of nested gages.

That probably explains why the results were so similar between our runs.

> we can test the model results with a variety of thresholds, but really we want to find a meaningful threshold in terms of how similar the basins are, right?

Yes, that's right. I think it makes sense. What would we use as the similarity metric?

Reply:
I'm not sure really - maybe the mean pearson correlation between all the pairs of streamflows in a nested group, or else a euclidean distance between basin characteristics or flow metrics? I don't know, we might have to talk about it. I wonder if this has been done before? I feel like someone has probably done this.

Reply:
@FlowChar any thoughts on this similarity metric to select a threshold for nestedness, Ken?

Reply:
> The result is about the same. Do you know how many nested gages are in each region? That could be a helpful number to have for each region

Here is a table with the total number of gages, number of distinct nested groups and number of nested gages in each region for a nestedness_threshold of 0.5.  While I was doing this, I noticed there are a number of nested groups that are split between different cluster regions, in which case they were not included in this table, but is interesting because we are assuming that nestedness means that they are similar enough that they would not be considered independent but they were apparently not similar enough to be placed in the same clusters. 

   flow_region cluster totaln nested_groups nested_gages
     midhigh       1    203             9                   18
      midhigh      2    308            13                  28
      midhigh      3    638            12                  24
      midhigh      4     89             4                     8
      midhigh      5    141             1                    2
         high         1    183             6                  12
         high         2    752            28                 58
         high         3    102             4                   8
         high         4    172             2                   4
        high          5    170             1                   2

Reply:
An idea on how to measure how similar a pair of gages are to each other in terms of flow is to normalize all monthly flows by drainage area (highs, lows, and everything in between should be captured) then do a simple stats test between the two distributions to see if they are pulled from the same population. Every pair of gages will have an associated DA overlap percentage. Bin the DA percentages and within each bin show the proportion of pairs that pass/reject the null hypothesis. Choose the bin as the threshold that has the largest ratio of rejections of the null hypothesis. I am suggesting a monthly time step so we get away from the 'lagging' problem of trying to compare flows at two different locations on the same day. (Note: for the GAGES 2.1 REF gages, I have all the monthly values already calculated for all years of records that can be shared).

Reply:
> Here is a table...

Thanks! I think this would be useful to keep as a target (e.g., write the table to a csv file).

It seems like 2-3 nested gages is most common, so that explains why we do not see much of a difference for our model results.

> there are a number of nested groups that are split between different cluster regions...

That is interesting! These could be gages on the boundary between classification into one cluster or another. Did you map out where these were located? It's also possible that the flow generation process in headwaters is different than events further downstream.

Reply:
@slevin75: do you want to try doing what Ken suggested to evaluate a nestedness threshold?